src/pybind/mgr/cephadm/services/services_map.py

```diff
@@ -0,0 +1,19 @@
+from typing import TYPE_CHECKING, Type
+
+if TYPE_CHECKING:
+    from cephadm.module import CephadmOrchestrator
+    from .cephadmservice import CephadmService
+
+registred_services: dict[str, Type["CephadmService"]] = {}
+cephadm_services: dict[str, "CephadmService"] = {}
```

Comment:
@phlogistonjohn I've done some refactoring giving a class entity to the services registry and adding some automation around services class discovery (because the decorator is only called when you import the service class). With the latest code services classes are discovered automatically so users shouldn't bother themselves importing them to perform the registration. Plz, have a look and let me know your opinion. 

Overall, my idea is to try seperating little by little the functionality of cephadm services on the services. We sill have the `mgr` dependency, which is not a good thing as in fact the `self.mgr` has be come kind of a big global variable which end up abused to get access to other parts of the code in general hiding the real dependencies that each class has. I think we have to git rid of that little by little as we do more refactoring. The final goal is to have a modular code where dependencies of each class/module are clear... but there's still a long way till we achieve that :)